utils/tools.go

```diff
@@ -200,3 +200,29 @@ func ConvertIntToByte32(n int) ([32]byte, error) {
 func ConvertByte32ToInt(nb [32]byte) int {
 	return int(binary.BigEndian.Uint64(nb[:]))
 }
+
+// ConvertProofForEthereum converts a proof to 32 byte format needed by ethereum
+func ConvertProofForEthereum(sortedHashes [][]byte) ([][32]byte, error) {
+	var property [][32]byte
+	for _, hash := range sortedHashes {
+		hash32, err := SliceToByte32(hash)
+		if err != nil {
+			return nil, err
+		}
+		property = append(property, hash32)
+	}
+
+	return property, nil
+}
+
+// GenerateHashPair generates a preimage and hash pair
+func GenerateHashPair(preimageSize int) (preimage, hash []byte, err error) {
+	preimage = RandomSlice(preimageSize)
+	h := sha256.New()
```

Comment:
hash function should be only called inside the crypto package.

Let's generate a sha256 function in that package.